doc/sphinx/advanced_methods.rst

```diff
@@ -1915,52 +1932,63 @@ particles). There exists a one to one mapping of the expressions in the
+Multiple reactions for the insertions of different types can be added to the same ``WidomInsertion`` instance. 
+Measuring the excess chemical potential using the insertion method is done via calling ``widom.measure_excess_chemical_potential(0)``. 
+The next insertion reaction can then be adressed with ``widom.measure_excess_chemical_potential(1)``. 
+Be aware that the implemented method only works for the canonical ensemble. If the numbers of particles fluctuate (i.e. in a semi grand canonical simulation) one has to adapt the formulas from which the excess chemical potential is calculated! This is not implemented. Also in a isobaric-isothermal simulation (NPT) the corresponding formulas for the excess chemical potentials need to be adapted. This is not implemented.
 
-    .. math::
+The implementation can also deal with the simultaneous insertion of multiple particles and can therefore measure the change of excess free energy of multiple particles like e.g.:
 
-       \mathrm{H_3O^+ + OH^- \rightleftharpoons\ \emptyset} \,,
+.. math::
 
-    with reaction constant 1/K. K is given implicitly as a function of the apparent dissociation
-    constant :math:`K_w=10^{-14} \rm{mol^2/l^2}=x\cdot \rm{1/(\sigma^3)^2}` such that the dimensionless is
-    :math:`K=(x\cdot \rm{1/(\sigma^3)^2})/(\beta P^0)^{\overline{\nu}}` with
-    :math:`\overline{\nu}=2` for the dissociation reaction and where x is
-    the value of the apparent dissociation constant that is converted from
-    :math:`\rm{mol^2/l^2}` to a number density in :math:`1/(\sigma^3)^2`,
-    where :math:`\sigma` is the simulation length unit. If :math:`\beta` and
-    :math:`P^0` are provided in simulation units this will make :math:`K`
-    dimensionless. As a test for the autodissociation of water a big
-    simulation box can be set up and the autodissociation reaction can be
-    performed. Then the box should fill with the correct number of protons
-    and hydroxide ions (check for the number of protons and hydroxide ions
-    in the given simulation volume and compare this to the expected value at
-    pH 7). Further the :math:`pK_w=14` should be reproduced -also in the
-    case of an initial excess of acid or base in the simulation box. Note
-    that this only works for big enough volumes.
+   \mu^\mathrm{ex, pair}:=\Delta F^\mathrm{ex, pair}:= F^\mathrm{ex}(N_1+1, N_2+1,V,T)-F^\mathrm{ex}(N_1, N_2 ,V,T)=-kT \ln \left(\frac{1}{V^2} \int_V \int_V d^3r_{N_1+1} d^3 r_{N_2+1} \langle \exp(-\beta \Delta E_\mathrm{pot}) \rangle_{N_1, N_2} \right)=-\kT \ln \left( \frac{1}{V^2} \int_V \int_V d^3r_{N_1+1} d^3 r_{N_2+1} \langle \exp(-\beta \Delta E_\mathrm{pot}) \rangle_N\right),
 
-Widom Insertion (for homogeneous systems)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Note that the measurement involves three averages: the ensemble average and the two volume averages. 
```

Comment:
a volume average is caluclated via f_V(t):=\frac{1}{V}int_V f(\vec{r},t) d^3r. This is spacial averaging of a quantity f.
The ensemble and volume average average of that quantity f could on the other hand be obtained via the time average (if the system is ergodic) <f_V>=1/T \int_0^T f_V(t)

Reply:
I disagree. The _ensemble average_ is by definition an average over all possible configurations in a given ensemble. 
Only if a given quantity can be defined locally, and if the system is sufficiently big, then the ensemble average can be approximated by averaging over different volume elements.

Only if the system is ergodic, then it can be approximated by time-averaging.

The term _volume average_ is not established in statistical mechanics (to my knowledge) and should not be used, or should be clearly defined before it is used. Otherwise, it does not explain anything to the reader.

Reply:
Please have a look at the above formula (see sphinx doc). It involves averages "over the position of particles N_1+1 and N_2+1" (this is how Frenkel calls these "volume averages", below eq. 7.2.5 in his book)... I do not mind calling them "averages over the position of particle N_1+1 and N_2+1", which is what i call a volume average.

Reply:
Some authors use the term "MC time". If you do not want to use the term _time average_, then we should think of a different term. Because _ensemble average_ is a general term that includes all kinds of averaging that one might perform within an ensemble.